Thanks for the explanation Chris. Yeah even I felt that 10s on it's own is a high enough timeout value. I looked at the logs again within the 436 folder above but what I see is that these [lines of code](https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/connect/connect_distributed_test.py#L475-L497) are getting executed, as can be seen in the logs 

```
[DEBUG - 2024-05-08 21:10:09,370 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers GET None
[DEBUG - 2024-05-08 21:10:09,508 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers GET response: 200
[DEBUG - 2024-05-08 21:10:09,508 - connect_distributed_test - test_dynamic_logging - lineno:475]: Listed all loggers via REST API: {'org.reflections': {'level': 'ERROR', 'last_modified': None}, 'root': {'level': 'DEBUG', 'last_modified': None}}
[DEBUG - 2024-05-08 21:10:09,508 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers/org.reflections GET None
[DEBUG - 2024-05-08 21:10:09,526 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers/org.reflections GET response: 200
[DEBUG - 2024-05-08 21:10:09,527 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers/org.reflections PUT {'level': 'INFO'}
[DEBUG - 2024-05-08 21:10:09,553 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers/org.reflections PUT response: 200
[DEBUG - 2024-05-08 21:10:09,553 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers GET None
[DEBUG - 2024-05-08 21:10:09,568 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers GET response: 200
[DEBUG - 2024-05-08 21:10:09,568 - connect_distributed_test - _loggers_are_set - lineno:619]: Read loggers on <ducktape.cluster.cluster.ClusterNode object at 0x7fb6526bed68> from Connect REST API: {'org.reflections': {'level': 'INFO', 'last_modified': 1715202609537}, 'root': {'level': 'DEBUG', 'last_modified': None}}
[DEBUG - 2024-05-08 21:10:09,568 - connect - _rest - lineno:265]: Kafka Connect REST request: worker10 http://10.140.57.125:8083/admin/loggers GET None
[DEBUG - 2024-05-08 21:10:09,729 - connect - _rest - lineno:267]: http://10.140.57.125:8083/admin/loggers GET response: 200
[DEBUG - 2024-05-08 21:10:09,729 - connect_distributed_test - _loggers_are_set - lineno:619]: Read loggers on <ducktape.cluster.cluster.ClusterNode object at 0x7fb6526bec88> from Connect REST API: {'org.reflections': {'level': 'ERROR', 'last_modified': None}, 'root': {'level': 'DEBUG', 'last_modified': None}}
[DEBUG - 2024-05-08 21:10:09,729 - connect - _rest - lineno:265]: Kafka Connect REST request: worker24 http://10.140.39.52:8083/admin/loggers GET None
[DEBUG - 2024-05-08 21:10:09,883 - connect - _rest - lineno:267]: http://10.140.39.52:8083/admin/loggers GET response: 200
[DEBUG - 2024-05-08 21:10:09,883 - connect_distributed_test - _loggers_are_set - lineno:619]: Read loggers on <ducktape.cluster.cluster.ClusterNode object at 0x7fb6526beba8> from Connect REST API: {'org.reflections': {'level': 'ERROR', 'last_modified': None}, 'root': {'level': 'DEBUG', 'last_modified': None}}
[DEBUG - 2024-05-08 21:10:09,883 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers/root?scope=cluster PUT {'level': 'DEBUG'}
[DEBUG - 2024-05-08 21:10:09,898 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers/root?scope=cluster PUT response: 204
[DEBUG - 2024-05-08 21:10:09,899 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers/root?scope=cluster PUT {'level': 'INFO'}
[DEBUG - 2024-05-08 21:10:09,914 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers/root?scope=cluster PUT response: 204
[DEBUG - 2024-05-08 21:10:09,915 - connect - _rest - lineno:265]: Kafka Connect REST request: worker4 http://10.140.42.37:8083/admin/loggers GET None
[DEBUG - 2024-05-08 21:10:09,926 - connect - _rest - lineno:267]: http://10.140.42.37:8083/admin/loggers GET response: 200
```

It is only after this line that I see the assert getting stuck on 

```
[DEBUG - 2024-05-08 21:10:09,926 - connect_distributed_test - _loggers_are_set - lineno:619]: Read loggers on <ducktape.cluster.cluster.ClusterNode object at 0x7fb6526bed68> from Connect REST API: 
``` 

and printing large blurbs of lines multiple times, and eventually failing with 

```
[INFO  - 2024-05-08 21:11:06,638 - runner_client - log - lineno:310]: RunnerClient: kafkatest.tests.connect.connect_distributed_test.ConnectDistributedTest.test_dynamic_logging: FAIL: AssertionError()
Traceback (most recent call last):
  File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/lib/python3.7/site-packages/ducktape/tests/runner_client.py", line 184, in _do_run
    data = self.run_test()
  File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/lib/python3.7/site-packages/ducktape/tests/runner_client.py", line 262, in run_test
    return self.test_context.function(self.test)
  File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/connect/connect_distributed_test.py", line 500, in test_dynamic_logging
    assert self._loggers_are_set(new_level, request_time, namespace, workers=[worker])
AssertionError
```
While I agree that the `/connectors` query might not be the best way to assert that the worker is well and truly up, in this case I would like to believe that the workers was up.

--- a/tests/kafkatest/tests/connect/connect_distributed_test.py
+++ b/tests/kafkatest/tests/connect/connect_distributed_test.py
@@ -21,7 +21,7 @@ from ducktape.cluster.remoteaccount import RemoteCommandError
 
 from kafkatest.services.zookeeper import ZookeeperService
 from kafkatest.services.kafka import KafkaService, config_property, quorum, consumer_group
-from kafkatest.services.connect import ConnectDistributedService, VerifiableSource, VerifiableSink, ConnectRestError, MockSink, MockSource
+from kafkatest.services.connect import ConnectDistributedService, ConnectServiceBase, VerifiableSource, VerifiableSink, ConnectRestError, MockSink, MockSource
 from kafkatest.services.console_consumer import ConsoleConsumer
 from kafkatest.services.security.security_config import SecurityConfig
 from kafkatest.version import DEV_BRANCH, LATEST_2_3, LATEST_2_2, LATEST_2_1, LATEST_2_0, LATEST_1_1, LATEST_1_0, LATEST_0_11_0, LATEST_0_10_2, LATEST_0_10_1, LATEST_0_10_0, LATEST_0_9, LATEST_0_8_2, KafkaVersion
@@ -468,7 +468,7 @@ class ConnectDistributedTest(Test):
 
         self.setup_services(num_workers=3)
         self.cc.set_configs(lambda node: self.render("connect-distributed.properties", node=node))
-        self.cc.start()
+        self.cc.start(mode=ConnectServiceBase.STARTUP_MODE_JOIN)
 
         worker = self.cc.nodes[0]
         initial_loggers = self.cc.get_all_loggers(worker)
@@ -633,9 +633,8 @@ class ConnectDistributedTest(Test):
     def _wait_for_loggers(self, level, request_time, namespace, workers=None):
         wait_until(
             lambda: self._loggers_are_set(level, request_time, namespace, workers),
-            # This should generally be super quick--just a write+read of the config topic, which workers are constantly polling,
-            # but the timeout is still set to a slightly higher value.
-            timeout_sec=30,
+            # This should be super quick--just a write+read of the config topic, which workers are constantly polling
+            timeout_sec=10,
             err_msg="Log level for namespace '" + namespace + "'  was not adjusted in a reasonable amount of time."
         )
 

